optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverFactory.java

```diff
@@ -175,6 +177,11 @@ protected RandomFactory buildRandomFactory(EnvironmentMode environmentMode_) {
         return phaseList;
     }
 
+    // required so we can set KieRuntimeBuilder in Quarkus
+    public void setKieBaseExtractor(KieBaseExtractor kieBaseExtractor) {
+        this.kieBaseExtractor = kieBaseExtractor;
+    }
```

Comment:
(important) How we past information from outside the box inside somewhere deep in the box needs to be consistent and scalable. This is not it. Let's discuss a better way.

Normally, the ScoreDirectorFactoryConfig is the way to pass such information.